pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -225,10 +237,10 @@ type HTTPSDConfig struct {
 // +k8s:openapi-gen=true
 type KubernetesSDConfig struct {
 	// Role of the Kubernetes entities that should be discovered.
-	// Currently the only supported role is "Node".
-	// +kubebuilder:validation:Enum=Node
 	// +required
-	Role string `json:"role"`
+	Role K8SRole `json:"role"`
+	// Selector to select objects.
+	Selectors []K8SSelectorConfig `json:"selectors,omitempty"`
```

Comment:
there can be only one item in the list for the same role.
```suggestion
	// +optional
	// +listType=map
	// +listMapKey=role
	Selectors []K8SSelectorConfig `json:"selectors,omitempty"`
```

Reply:
@simonpasquier  do we want to specify a default here since selectors is not required field?
```
 main_test.go:184: initialize ScrapeConfig v1alpha1 CRD: create CRD scrapeconfigs on the apiserver: create CRD ScrapeConfig: CustomResourceDefinition.apiextensions.k8s.io "scrapeconfigs.monitoring.coreos.com" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[kubernetesSDConfigs].items.properties[selectors].items.properties[role].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
```